mechanics/tools/gdocs-migrate/client.go

```diff
@@ -0,0 +1,156 @@
+	if err != nil {
+		return nil, err
+	}
+	return &File{
+		id:       r.Id,
+		name:     r.Name,
+		driveID:  driveId,
+		isFolder: true,
+	}, nil
+}
+
+func loadAllFiles(forest *Forest, srv *drive.Service, drive File) {
+	forest.Add(drive)
+	var pageToken *string
+	for pageToken == nil || *pageToken != "" {
+		request := srv.Files.List().PageSize(1000).
+			Corpora("drive").
+			IncludeItemsFromAllDrives(true).
+			SupportsAllDrives(true).
+			DriveId(drive.id).
+			Fields("nextPageToken", "files(id, name, parents, mimeType, trashed)").
+			OrderBy("folder,modifiedTime desc,name")
+		if pageToken != nil {
+			request.PageToken(*pageToken)
+		}
+		r, err := request.Do()
+		if err != nil {
+			log.Fatal("Unable to retrieve files:", err)
+		}
+		pageToken = &r.NextPageToken
+		for _, i := range r.Files {
+			if i.Trashed {
+				// Ignore trashed files and folders.
+				continue
+			}
+			forest.Add(File{
+				id:       i.Id,
+				name:     i.Name,
+				parentID: i.Parents[0],
+				driveID:  drive.id,
+				isFolder: i.MimeType == FolderMimeType,
+				mimeType: i.MimeType,
+			})
+		}
+		// Print progress.
+		log.Printf("Loaded %d files, %d folders\n", len(forest.GetFiles()), forest.FolderCount())
+	}
+}
+
+func createDir(srv *drive.Service, parentDir File, name string) (*File, error) {
+	f, err := srv.Files.Create(&drive.File{
+		Name:     name,
+		Parents:  []string{parentDir.id},
+		DriveId:  parentDir.driveID,
+		MimeType: FolderMimeType,
+	}).SupportsAllDrives(true).Do()
+	if err != nil {
+		return nil, err
+	}
+	return &File{
+		id:       f.Id,
+		name:     f.Name,
+		parentID: parentDir.id,
+		driveID:  parentDir.driveID,
+		isFolder: true,
+	}, nil
+}
+
+func deleteFile(srv *drive.Service, fileId string) error {
+	return srv.Files.Delete(fileId).SupportsAllDrives(true).Do()
+}
+
+func prependAuthor(author *drive.User, content string) string {
+	return author.DisplayName + ": " + content
+}
+
+func sanitizeEmail(content string) string {
+	// Use a homoglyph of @ to make sure people aren's getting emails
+	return strings.ReplaceAll(content, "@", "\uff20")
+}
+
+// reformat the time from RFC3339 for brevity
+func decorateTime(rfc3339 string) string {
+	t, _ := time.Parse(time.RFC3339, rfc3339)
+	return t.UTC().Format("2006-Jan-02 23:04")
+}
+
+func decorateComment(input *drive.Comment) *drive.Comment {
+	comment := *input
+	comment.Replies = nil
+	comment.Content = fmt.Sprintf("[%s] %s: %s",
+		decorateTime(input.CreatedTime), input.Author.DisplayName, sanitizeEmail(input.Content))
+	return &comment
+}
+
+func decorateReply(input *drive.Reply) *drive.Reply {
+	reply := *input
+	reply.Content = fmt.Sprintf("[%s] %s: %s",
+		decorateTime(input.CreatedTime), input.Author.DisplayName, sanitizeEmail(input.Content))
+	return &reply
+}
+
+func copyFile(srv *drive.Service, source File, dstDir File) (*File, error) {
+	f, err := srv.Files.Copy(source.id, &drive.File{
+		Name:    source.name,
+		Parents: []string{dstDir.id},
+		DriveId: dstDir.driveID,
```

Comment:
It looks like you may be able to set `modifiedTime` as well.

https://developers.google.com/drive/api/v3/reference/files/copy

Reply:
Done